iscsi-scst/usr/iscsid.c

```diff
@@ -498,6 +498,291 @@ static int init_conn_session_params(struct connection *conn)
 	return res;
 }
 
+static bool valid_ascii_name_char(char ch)
+{
+
+	/*
+	 * From rtf3722
+	 *
+	 * For those using only ASCII characters (U+0000 to U+007F), the
+	 *   following characters are allowed:
+	 *
+	 *   -  ASCII dash character ('-' = U+002d)
+	 *   -  ASCII dot character ('.' = U+002e)
+	 *   -  ASCII colon character (':' = U+003a)
+	 *   -  ASCII lower-case characters ('a'..'z' = U+0061..U+007a)
+	 *   -  ASCII digit characters ('0'..'9' = U+0030..U+0039)
+	 */
+
+	if ((ch == '-') || (ch == '.') || (ch == ':') ||
+	    ((ch >= 'a') && (ch <= 'z')) ||
+	    ((ch >= '0') && (ch <= '9')))
+		return true;
+
+	log_error("Name contains invalid character: %c", ch);
+	return false;
+}
+
+static bool valid_utf8_name_char(long ch)
+{
+	/*
+	 * From rfc3722 6.  Prohibited Output
+	 *
+	 *   This profile specifies prohibiting using the following tables from
+	 *   [RFC3454].  Characters appearing within these tables MUST NOT be used
+	 *   within an iSCSI name.
+	 *
+	 *      Table C.1.1
+	 *      Table C.1.2
+	 *      Table C.2.1
+	 *      Table C.2.2
+	 *      Table C.3
+	 *      Table C.4
+	 *      Table C.5
+	 *      Table C.6
+	 *      Table C.7
+	 *      Table C.8
+	 *      Table C.9
+	 */
+
+	/* C.1.1 ASCII space characters */
+	if (ch == 0x0020) {
+		log_error("Invalid UTF-8: %lX (ASCII space characters)", ch);
+		return false;
+	}
+
+	/* C.1.2 Non-ASCII space characters */
+	if ((ch == 0x00A0) ||	/* NO-BREAK SPACE */
```

Comment:
This is an opportunity for the `switch` statement:

```
switch (ch) {
case 0x00A0:
case 0x1680:
...
return false;
}
```
The `switch` statement is more effective than `if` when you have many cases.
However, it's a suggestion, not a strict requirement, so if you prefer using `if` here, just resolve the thread.